src/ProxyClient/Fastly.php

```diff
@@ -0,0 +1,130 @@
+<?php
+
+/*
+ * This file is part of the FOSHttpCache package.
+ *
+ * (c) FriendsOfSymfony <http://friendsofsymfony.github.com/>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace FOS\HttpCache\ProxyClient;
+
+use FOS\HttpCache\ProxyClient\Invalidation\ClearCapable;
+use FOS\HttpCache\ProxyClient\Invalidation\PurgeCapable;
+use FOS\HttpCache\ProxyClient\Invalidation\TagCapable;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Fastly HTTP cache invalidator.
+ *
+ * @author Simone Fumagalli <simone.fumagalli@musement.com>
+ */
+class Fastly extends HttpProxyClient implements TagCapable, PurgeCapable, ClearCapable
+{
+    /**
+     * @internal
+     */
+    const HTTP_METHOD_PURGE = 'PURGE';
+
+    /**
+     * {@inheritdoc}
+     *
+     * @see https://docs.fastly.com/api/purge#purge_db35b293f8a724717fcf25628d713583
+     */
+    public function invalidateTags(array $tags)
+    {
+        $headers = [
+            'Fastly-Key' => $this->options['authentication_token'],
+            'Accept' => 'application/json'
+        ];
+
+        if (true === $this->options['soft_purge']) {
+            $headers['Fastly-Soft-Purge'] = 1;
+        }
+
+        foreach (\array_chunk($tags, 256) as $tagchunk) {
+            $this->queueRequest(
+                Request::METHOD_POST,
+                sprintf("/service/%s/purge", $this->options['service_identifier']),
+                $headers + [
+                    'Surrogate-Key' => implode(' ', $tagchunk),
```

Comment:
You may want to change the surrogate keys to JSON payload instead of sending them as a header. 

https://github.com/fastly/fastly-magento2/blob/master/Model/Api.php#L197

Reply:
Due to escaping? It will still be limited to 256 tags right?

Reply:
It's still limited to 256. If payload is a problem you can just send it as a header. It was just a little bit cleaner.

Reply:
as long as we are BC (which we should be able to) i think it makes sense to add an optional body to `queueRequest`. parameter ordering will be a bit "funny" because of the validateHost flag we already have, but so be it...